src/overlays/actors/ovl_En_Bili/z_en_bili.c

```diff
@@ -46,69 +59,705 @@ static ColliderCylinderInit D_809C1640 = {
+    this->actionFunc = EnBili_ApproachPlayer;
+}
+
+void EnBili_SetupSetNewHomeHeight(EnBili* this) {
+    Animation_PlayLoop(&this->skelAnime, &D_060000A4);
+    this->timer = 96;
+    this->actor.speedXZ = 0.9f;
+    this->collider.base.atFlags |= AT_ON;
+    this->actionFunc = EnBili_SetNewHomeHeight;
+    this->actor.home.pos.y = this->actor.world.pos.y;
+}
+
+void EnBili_SetupFlee(EnBili* this) {
+    if (this->skelAnime.animation != &D_060000A4) {
+        Animation_PlayLoop(&this->skelAnime, &D_060000A4);
+    }
+
+    this->actor.world.rot.y = Actor_WorldYawTowardPoint(&this->actor, &this->collider.base.ac->prevPos) + 0x8000;
+    this->actor.world.rot.x = Actor_WorldPitchTowardPoint(&this->actor, &this->collider.base.ac->prevPos);
+    this->actionFunc = EnBili_Flee;
+    this->actor.speedXZ = 5.0f;
+}
+
+/**
+ * Used for both fire damage and generic damage
+*/
+void EnBili_SetupBurnt(EnBili* this) {
+    if (this->actionFunc == EnBili_Climb) {
+        Animation_PlayLoop(&this->skelAnime, &D_060000A4);
+    }
+
+    this->timer = 20;
+    this->collider.base.atFlags &= ~AT_ON;
+    this->collider.base.acFlags &= ~AC_ON;
+    this->actor.flags |= 0x10;
+    this->actor.speedXZ = 0.0f;
+    func_8003426C(&this->actor, 0x4000, 0xC8, 0x2000, 0x14);
+    this->actionFunc = EnBili_Burnt;
+}
+
+void EnBili_SetupDie(EnBili* this) {
+    this->timer = 18;
+    this->actor.flags &= ~1;
+    this->actionFunc = EnBili_Die;
+    this->actor.speedXZ = 0.0f;
+}
+
+/**
+ * Falls to ground
+*/
+void EnBili_SetupStunned(EnBili* this) {
+    this->timer = 80;
+    this->collider.info.bumper.effect = 0;
+    this->actor.gravity = -1.0f;
+    this->actor.speedXZ = 0.0f;
+    func_8003426C(&this->actor, 0, 0x96, 0x2000, 0x50);
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_GOMA_JR_FREEZE);
+    this->collider.base.atFlags &= ~AT_ON;
+    this->actionFunc = EnBili_Stunned;
+}
+
+void EnBili_SetupFrozen(EnBili* this, GlobalContext* globalCtx) {
+    s32 i;
+    Vec3f effectPos;
+
+    if (!(this->actor.flags & 0x8000)) {
+        this->actor.gravity = -1.0f;
+    }
+
+    this->actor.velocity.y = 0.0f;
+    effectPos.y = this->actor.world.pos.y - 15.0f;
+
+    for (i = 0; i < 8; i++) {
+
+        effectPos.x = this->actor.world.pos.x + ((i & 1) ? 7.0f : -7.0f);
+        effectPos.y += 2.5f;
+        effectPos.z = this->actor.world.pos.z + ((i & 4) ? 7.0f : -7.0f);
+
+        EffectSsEnIce_SpawnFlyingVec3f(globalCtx, &this->actor, &effectPos, 150, 150, 150, 250, 235, 245, 255,
+                                       (Rand_ZeroOne() * 0.2f) + 0.7f);
+    }
+
+    this->actor.speedXZ = 0.0f;
+    func_8003426C(&this->actor, 0, 0x96, 0x2000, 0xA);
+    this->collider.base.atFlags &= ~AT_ON;
+    this->collider.base.acFlags &= ~AC_ON;
+    this->timer = 300;
+    this->actionFunc = EnBili_Frozen;
+}
+
+// Miscellaneous
+
+/**
+ * Changes the texture displayed on the oral arms limb using the current frame.
+*/
+void EnBili_UpdateOralArmsIndex(EnBili* this) {
```

Comment:
i still think that tentacles are going to be much easier for the average reader to understand, even if its technically less correct lol